catkit2/services/accufiz_interferometer/accufiz_interferometer.py

```diff
@@ -0,0 +1,270 @@
+        if (not self.save_h5) and os.path.exists(local_file_path):
+            os.remove(local_file_path)
+
+        return image
+
+    @staticmethod
+    def convert_h5_to_fits(filepath, rotate, fliplr, img, mask, wavelength=632.8, create_fits=False):
+        """
+        Convert HDF5 data to FITS format and process image data.
+
+        Args:
+            filepath (str): Filepath for the HDF5 data.
+            rotate (int): Rotation angle in degrees.
+            fliplr (bool): If True, flip the image horizontally.
+            img (numpy.ndarray): Image data to be processed.
+            mask (numpy.ndarray): Mask data to be applied.
+            wavelength (float, optional): Wavelength for scaling. Defaults to 632.8 nm.
+            create_fits (bool, optional): If True, save the processed image as a FITS file.
+
+        Returns:
+            numpy.ndarray: Processed image data.
+        """
+        filepath = filepath if filepath.endswith(".h5") else f"{filepath}.h5"
+        fits_filepath = f"{os.path.splitext(filepath)[0]}.fits"
+
+        mask = np.array(h5py.File(filepath, 'r').get('measurement0').get('Detectormask', 1))
+        img = np.array(h5py.File(filepath, 'r').get('measurement0').get('genraw').get('data')) * mask
+
+        if create_fits:
+            fits.PrimaryHDU(mask).writeto(fits_filepath, overwrite=True)
+
+        radiusmask = np.int64(np.sqrt(np.sum(mask) / math.pi))
+        center = ndimage.measurements.center_of_mass(mask)
+
+        image = np.clip(img, -10, +10)[np.int64(center[0]) - radiusmask:np.int64(center[0]) + radiusmask - 1,
+                                       np.int64(center[1]) - radiusmask: np.int64(center[1]) + radiusmask - 1]
+
+        # Apply the rotation and flips.
+        image = rotate_and_flip_image(image, rotate, fliplr)
+
+        # Convert waves to nanometers.
+        image = image * wavelength
+
+        if create_fits:
+            fits_hdu = fits.PrimaryHDU(image)
+            fits_hdu.writeto(fits_filepath, overwrite=True)
+
+        return image
+
+    def main(self):
+        """
+        Main loop to manage data acquisition and processing.
+        """
+        while not self.should_shut_down:
+            if self.should_be_acquiring.wait(0.05):
+                self.acquisition_loop()
+
+    def acquisition_loop(self):
+        """
+        Handle continuous data acquisition while the service is running.
+        """
+        try:
+            self.is_acquiring.submit_data(np.array([1], dtype='int8'))
+
+            while self.should_be_acquiring.is_set() and not self.should_shut_down:
+                img = self.take_measurement()
+
+                has_correct_parameters = np.allclose(self.images.shape, img.shape)
+
+                if not has_correct_parameters:
+                    self.images.update_parameters('float32', img.shape, 20)
+
+                self.images.submit_data(img.astype('float32'))
+                time.sleep(0.05)
```

Comment:
Just noticing that right now. Is there a specific reason you'd want a sleep here?

Reply:
No particular reason. Especially since I've migrated to the camera interface loop approach. Good catch!